web/settings/clients_usage.go

```diff
@@ -0,0 +1,81 @@
+package settings
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/cozy/cozy-stack/model/oauth"
+	"github.com/cozy/cozy-stack/model/permission"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/jsonapi"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/labstack/echo/v4"
+)
+
+type apiClientsUsage struct {
+	Limit         *int `json:"limit,omitempty"`
+	Count         int  `json:"count"`
+	LimitReached  bool `json:"limitReached"`
+	LimitExceeded bool `json:"limitExceeded"`
+}
+
+func (j *apiClientsUsage) ID() string                             { return consts.DiskUsageID }
+func (j *apiClientsUsage) Rev() string                            { return "" }
+func (j *apiClientsUsage) DocType() string                        { return consts.Settings }
+func (j *apiClientsUsage) Clone() couchdb.Doc                     { return j }
+func (j *apiClientsUsage) SetID(_ string)                         {}
+func (j *apiClientsUsage) SetRev(_ string)                        {}
+func (j *apiClientsUsage) Relationships() jsonapi.RelationshipMap { return nil }
+func (j *apiClientsUsage) Included() []jsonapi.Object             { return nil }
+func (j *apiClientsUsage) Links() *jsonapi.LinksList {
+	return &jsonapi.LinksList{Self: "/settings/clients-usage"}
+}
+
+// Settings objects permissions are only on ID
+func (j *apiClientsUsage) Fetch(field string) []string { return nil }
+
+// checkAccessToClientsUsage validates the access control for the clients-usage
+// route. It checks if there is an explicit permission on this document, but
+// also allow every request from the flagship app as this route is used by it to
+// decide whether it can finalize its onboarding or not.
+func checkAccessToClientsUsage(c echo.Context, result *apiClientsUsage) error {
```

Comment:
I think `return middlewares.Allow(c, permission.GET, result)` is enough

Reply:
Do you think the `checkAccessToClientsUsage` function is still useful? I would inline it in the `clientsUsage` function